lib/perl/Genome/Model/ClinSeq.pm

```diff
@@ -249,6 +233,19 @@ sub _resolve_reference {
 sub __errors__ {
     my $self   = shift;
     my @errors = $self->SUPER::__errors__;
+
+    for my $model (qw(wgs_model exome_model)) {
+        if ($self->$model) {
+            unless ($self->$model->isa('Genome::Model::SomaticVariation') || $self->$model->isa('Genome::Model::SomaticValidation')) {
```

Comment:
Making the `SomaticInterface` into a role sounds like a good idea. I'm willing to tackle the necessary UR changes. @brummett can you provide some pointers of what would need to change  for role type checking to work?